service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantController.java

```diff
@@ -39,13 +40,24 @@
     private final PbaNumberService pbaNumberService;
     private final OrganisationService organisationService;
     private final ValidateEmailService validateEmailService;
+    private final FeatureToggleService featureToggleService;
 
     @PostMapping("/about-to-start")
-    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(
+        @RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
         CaseData caseData = getCaseData(caseDetails);
-
-        Organisation organisation = organisationService.findOrganisation().orElse(Organisation.builder().build());
+        Organisation organisation;
+
+        if (featureToggleService.isRetrieveManagedOrganisation()
```

Comment:
as we discussed with Susheel, this behaviour is common for both EPS and MLA, so we dont have to check what kind of outsourcing it is.
if you have privet method like:
```
 private Optional<Organisation> getOrganisation(CaseData caseData) {
        if (featureToggleService.isRetrieveManagedOrganisation()) {
            String organisationId = caseData.getLocalAuthorityPolicy().getOrganisation().getOrganisationID();
            return organisationService.findManagedOrganisation(organisationId);
        } else {
            return organisationService.findOrganisation();
        }
    }

```
then your code can looks like:

 Organisation organisation = getOrganisation(caseData).orElse(Organisation.builder().build());

        caseDetails.getData()
            .put(APPLICANTS_PROPERTY, applicantService.expandApplicantCollection(caseData, organisation))

Reply:
TBH, i believe that old logic is not correct and we should use new one even for non outsourced cases

Reply:
I agree it would be tidier to just use the new endpoint for populating the fields. I think we'll need to do it this way for now since we need the toggle. Seems like something that should be revisited